database/seeds/ModelSeeders/UserSeeder.php

```diff
@@ -0,0 +1,108 @@
+            $this->command->info('Can\'t seed events, scores, events or favourite maps due to having no beatmap data.');
+        }
+
+        // Store some modifiers
+        $this->improvement_speeds = [
+            (rand(100, 110) / 100), // Fast Learner
+            (rand(100, 102) / 100), // Slow Learner
+            (rand(100, 115) / 100), // Genius / Multiaccounter :P
+        ];
+
+        // Create 10 users and their stats
+        factory(App\Models\User::class, 10)->create()->each(function ($u) {
+
+            // USER STATS
+            $common_countries = ['US', 'JP', 'CN', 'DE', 'TW', 'RU', 'KR', 'PL', 'CA', 'FR', 'BR', 'GB', 'AU'];
+            $country_code = $common_countries[array_rand($common_countries)];
+
+            $rank0 = rand(1, 500000);
+            $rank1 = rand(1, 500000);
+            $rank2 = rand(1, 500000);
+            $rank3 = rand(1, 500000);
+            $st = $u->statisticsOsu()->save(factory(App\Models\UserStatistics\Osu::class)->create(['country_acronym' => $country_code, 'rank' => $rank0, 'rank_score_index' => $rank0]));
+            $st1 = $u->statisticsOsu()->save(factory(App\Models\UserStatistics\Taiko::class)->create(['country_acronym' => $country_code, 'rank' => $rank1, 'rank_score_index' => $rank1]));
+            $st2 = $u->statisticsOsu()->save(factory(App\Models\UserStatistics\Fruits::class)->create(['country_acronym' => $country_code, 'rank' => $rank2, 'rank_score_index' => $rank2]));
+            $st3 = $u->statisticsOsu()->save(factory(App\Models\UserStatistics\Mania::class)->create(['country_acronym' => $country_code, 'rank' => $rank3, 'rank_score_index' => $rank3]));
+            // END USER STATS
+
+            // RANK HISTORY
+
+            // Create rank histories for all 4 modes
+            for ($c = 0; $c <= 3; $c++) {
+                switch ($c) {
+                    case 0: $rank = $st->rank; break;
+                    case 1: $rank = $st1->rank; break;
+                    case 2: $rank = $st2->rank; break;
+                    case 3: $rank = $st3->rank; break;
+                    default: $rank = $st->rank;
+                }
+
+                $hist = new App\Models\RankHistory;
+
+                $hist->mode = $c; // 0 = standard, 1 = taiko etc...
+
+                $play_freq = rand(10, 35); // How regulary the user plays (as a % chance per day)
+
+                $improvement_speed = $this->improvement_speeds[array_rand($this->improvement_speeds)];
+
+                // Start with current rank, and move down (back in time) to r0
+                $hist->r89 = $rank;
+
+                for ($i = 88; $i >= 0; $i--) {
+                    $r = 'r'.$i;
+                    $prev_r = 'r'.($i + 1);
+                    $prev_rank = $hist->$prev_r;
+                    // We wouldn't expect the user to improve every day
+                    $does_improve = $this->faker->boolean($play_freq);
+                    if ($does_improve === true) {
```

Comment:
I use variables like $does_improve and $extreme_improvement, because it makes it easier to understand the logic behind the code. This is what I would find easier to read.

For me the line  `if ($this->faker->boolean($playFreq) {` doesn't make as much sense as:

```
$does_improve = $this->faker->boolean($play_freq);
       if ($does_improve === true) {
```

The variable $does_improve actually explains the concept of what the random chance applies to, and for me this is easier to understand.